### PR TITLE
Fixed GenerateAssemblyPlugins and a hud bitmap widget.

### PR DESCRIPTION
**Pull request by craftycodie**
Fixed an issue where tag structure strings were not converted properly due to improperly handled length attribute. Fixed an issue where the scoreboard 'who_am_i' arrow didn't show in forge.

### CODE REVIEW

File: TagTool/Commands/Porting/PortTagCommand.HUD.cs [hunk 1]
```diff
@@ -93,6 +93,14 @@ private ChudDefinition ConvertChudDefinition(ChudDefinition chudDefinition)
 
                 for (int bitmapWidgetIndex = 0; bitmapWidgetIndex < chudDefinition.HudWidgets[hudWidgetIndex].BitmapWidgets.Count; bitmapWidgetIndex++)
                 {
+                    if (chudDefinition.HudWidgets[hudWidgetIndex].BitmapWidgets[bitmapWidgetIndex].Name.Value == 17796)
```
[the remark this answers is not in the document]
Comment (craftycodie) [reply TagTool/Commands/Porting/PortTagCommand.HUD.cs]: I'll cut it out, it's nothing significant. If I remember correctly it's just the budget string in forge or something like that.

[the remark this answers is not in the document]
Comment (CollinMcKinney) [reply TagTool/Commands/Porting/PortTagCommand.HUD.cs]:
```suggestion
(chudDefinition.HudWidgets[hudWidgetIndex].BitmapWidgets[bitmapWidgetIndex].Name == CacheContext.GetStringId("sandbox_budget");
```

[the remark this answers is not in the document]
Comment (craftycodie) [reply TagTool/Commands/Porting/PortTagCommand.HUD.cs]: That looks good, I'll update tomorrow, afk for the night.

[the remark this answers is not in the document]
Comment (craftycodie) [reply TagTool/Commands/Porting/PortTagCommand.HUD.cs]: Updated.

**Review by camden-smallwood-zz — APPROVED** [review]

**Pull request merged**